### ISSUE TITLE
Spinner: unable to tab to next field

### ISSUE THREAD

**Issue by jepsar** [primefaces/primefaces#7378]
**Reproducer**
https://www.primefaces.org/showcase/ui/input/spinner.xhtml
Focus a spinner and press tab

**Environment:**
 - PF Version: _10.0_, _11.0.0-SNAPSHOT_ (was working in 8)
 - Affected browsers: _Chrome, Firefox_ (probably all)

Note that `Shift+Tab` should also work.

Note that it is working in https://primefaces.org/primeng/showcase/#/inputnumber

**Comment by melloware** [comment]
MonkeyPatch: 
```javascript
if (PrimeFaces.widget.Spinner) {
    PrimeFaces.widget.Spinner.prototype.bindEvents = function() {
        var $this = this;

        this.jq.children('.ui-spinner-button')
            .on('mouseover.spinner', function() {
                $(this).addClass('ui-state-hover');
            })
            .on('mouseout.spinner', function() {
                $(this).removeClass('ui-state-hover ui-state-active');

                if ($this.timer) {
                    clearInterval($this.timer);
                }
            })
            .on('mouseup.spinner', function() {
                clearInterval($this.timer);
                $(this).removeClass('ui-state-active').addClass('ui-state-hover');
                $this.input.trigger('change');
            })
            .on('mousedown.spinner', function(e) {
                // only act on left click
                if (e.which !== 1) {
                    return;
                }
                var element = $(this),
                    dir = element.hasClass('ui-spinner-up') ? 1 : -1;

                element.removeClass('ui-state-hover').addClass('ui-state-active');

                if ($this.input.is(':not(:focus)')) {
                    $this.input.trigger('focus');
                }

                $this.repeat(null, dir);

                //keep focused
                e.preventDefault();
            });

        this.input.on('keydown.spinner', function(e) {
                var keyCode = $.ui.keyCode;

                switch (e.which) {
                    case keyCode.UP:
                        $this.spin(1);
                        break;

                    case keyCode.DOWN:
                        $this.spin(-1);
                        break;

                    case keyCode.ENTER:
                        $this.updateValue();
                        $this.format();
                        break;

                    case keyCode.BACKSPACE:
                    case keyCode.DELETE:
                    case keyCode.LEFT:
                    case keyCode.RIGHT:
                    case keyCode.TAB:
                        return;

                    default:
                        //do nothing
                        break;
                }

                /* Github #1964 do not allow minus */
                var isNegative = event.key === '-';
                if ($this.cfg.min >= 0 && isNegative) {
                    e.preventDefault();
                    return;
                }

                /* GitHub #5579 do not allow decimal separator for integers */
                var isDecimalSeparator = event.key === $this.cfg.decimalSeparator;
                if (isDecimalSeparator && $this.cfg.precision === 0) {
                    e.preventDefault();
                    return;
                }

                /* GitHub #5579 prevent non numeric characters and duplicate separators */
                var value = $(this).val();
                var isNumber = isFinite(event.key);
                var isThousandsSeparator = event.key === $this.cfg.thousandSeparator;
                if ((isNegative && value.indexOf('-') != -1) ||
                    (isDecimalSeparator && value.indexOf($this.cfg.decimalSeparator) != -1) ||
                    (isThousandsSeparator && value.indexOf($this.cfg.thousandSeparator) != -1)) {
                    e.preventDefault();
                    return;
                }

                if (!isNumber && !(isNegative || isDecimalSeparator || isThousandsSeparator)) {
                    e.preventDefault();
                    return;
                }
            })
            .on('keyup.spinner', function(e) {
                $this.updateValue();

                var keyCode = $.ui.keyCode;

                /* Github #2636 */
                var checkForIE = (PrimeFaces.env.isIE(11) || PrimeFaces.env.isLtIE(11)) && (e.which === keyCode.ENTER);

                if (e.which === keyCode.UP || e.which === keyCode.DOWN || checkForIE) {
                    $this.input.trigger('change');
                    $this.format();
                }
            })
            .on('blur.spinner', function(e) {
                $this.format();
            })
            .on('mousewheel.spinner', function(event, delta) {
                if ($this.input.is(':focus')) {
                    if (delta > 0)
                        $this.spin(1);
                    else
                        $this.spin(-1);

                    $this.input.trigger('change');

                    return false;
                }
            });
    }
}
```

**Comment by jepsar** [comment]
@melloware Should we label this as elite?

**Comment by melloware** [comment]
Nah i think they should bump to 11.0.0 for this fix.
